bytecode/src/function/instructions/add_wrapped.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::{
+    function::{parsers::*, Instruction, Opcode, Operation, Registers},
+    helpers::Register,
+    Program,
+    Value,
+};
+use snarkvm_circuits::{AddWrapped as AddWrappedCircuit, Literal, Parser, ParserResult};
+use snarkvm_utilities::{FromBytes, ToBytes};
+
+use core::fmt;
+use nom::combinator::map;
+use std::io::{Read, Result as IoResult, Write};
+
+/// Adds `first` with `second`, wrapping around on overflow, and storing the outcome in `destination`.
+pub struct AddWrapped<P: Program> {
+    operation: BinaryOperation<P>,
+}
+
+impl<P: Program> AddWrapped<P> {
+    /// Returns the operands of the instruction.
+    pub fn operands(&self) -> Vec<Operand<P>> {
+        self.operation.operands()
+    }
+
+    /// Returns the destination register of the instruction.
+    pub fn destination(&self) -> &Register<P> {
+        self.operation.destination()
+    }
```

Comment:
these 2 methods look like good candidates for a dedicated trait that would apply to all objects containing a `BinaryOperation`

Reply:
Yes, perhaps we could add these 2 methods to Operation. The way the Instruction enum is currently implemented, it does require all variants to implement these methods anyways. 

@howardwu Do you think this behavior should be captured under Operation or maybe under a separate trait?